modules/auxiliary/scanner/mysql/mysql_login.rb

```diff
@@ -149,6 +175,23 @@ def int_version(str)
     end
   end
 
+  # @param [Metasploit::Framework::LoginScanner::Result] result
+  # @param [::Mysql] client
+  # @return [Msf::Sessions::MYSQL]
+  def session_setup(result, client)
+    return unless (result && client)
 
+    rstream = client
```

Comment:
is this correct? I don't think rstream and client are supposed to be the same thing, pretty sure rstream is expected to be a Rex::Socket and then client would be your MySQL::Client?

Reply:
Yes, I have updated that to be the rex socket. In regards to `nil` working it was being handled as part of `start_session` where `my_session.rstream` was being passed.